include/la/avdecc/utils.hpp

```diff
@@ -285,13 +283,9 @@ class EnumBitfield final
 			}
 			return *this;
 		}
-		reference operator*() noexcept
+		value_type operator*() noexcept
```

Comment:
This one I'll have to check deeply!
This might be a breaking change since returning by value prevents modifying the value pointed by the iterator.

Reply:
I had thought about this for a while and decided it should not even be a visible change at all.
The value referenced is a member of the iterator itself (`_currentValue`) and not something it points to. None of the member function actually read this value, it is always recomputed relying on `_currentBitPosition`first . Therefore a reference to `_currentValue` should behave identical to a copy thereof.

Reply:
Hum yes you are probably right, I'll still have to check tomorrow with my unit tests if this is something desirable (to be able to alter the value from the *iterator). If not, then yes it'll be better to user a copy

Reply:
I took the time to re-read the class and it's purpose (wrote this a long time ago). The iterator was not meant to offer the possibility to change which enum is set or not.

So you are absolutely right, I should change it like you proposed.
The other advantage will be to cause a compilation error if someone tried to do something like 
> *it = val;
(which was possible with the current implementation, but leading to something probably not expected by the user).

Thanks again, nice catch.
(out of curiosity, did you had a compilation issue with that, or was it an extensive review of the class?)

Reply:
As the commit message says "This avoids a reinterpret_cast that broke C++'s strict aliasing rule."
So, yes this produced a warning.

Reply:
Very interesting, I'll definitely check my gcc version, cause I'm pretty sure it compiles as -Wall

Quick semi-unrelated note, since I had to make Hive compile for gcc last week, I already fixed all the warnings that *my* gcc version produced (private branch).
They are scheduled to be merged tomorrow as a new beta version will be prepared this week.

Reply:
Fresh head. I should remove operator-> as well from the iterator, we really don't want someone to be able to get an lvalue from the iterator and try to modify the value. The iterator was only meant as a read-only way to iterate through the bit that are set.
My gcc version is 8.3.0, so I'll just set the minimum requirements to 8.2.1

Reply:
We forgot something here.
The method should have been changed to **const**